### PR TITLE
chore: use revive linter

### PR DESCRIPTION
**Pull request by faddat**
This PR enables the revive linter, and should be merged after gci and gofumpt, since it contains them. 

- use gofmput to format code
- use gci to format imports
- reconfigure gci
- linter autofixes
- rearrange imports a little
- all changes other than unused-parameter
- use revive linter

## Contains

* #5707 
* #5708

### CODE REVIEW

File: modules/caddyhttp/ip_range.go [hunk 1]
```diff
@@ -86,16 +86,16 @@ func (s *StaticIPRange) GetIPRanges(_ *http.Request) []netip.Prefix {
 }
 
 // UnmarshalCaddyfile implements caddyfile.Unmarshaler.
-func (m *StaticIPRange) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+func (f *StaticIPRange) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
```
Comment (francislavoie) [inline modules/caddyhttp/ip_range.go]: Where does `f` come from? We typically use `m` to mean "module".

File: cmd/commandfuncs.go [hunk 2]
```diff
@@ -630,7 +630,7 @@ func AdminAPIRequest(adminAddr, method, uri string, headers http.Header, body io
 	if err != nil {
 		return nil, fmt.Errorf("making request: %v", err)
 	}
-	if parsedAddr.IsUnixNetwork() {
+	if parsedAddr.IsUnixNetwork() { //nolint:revive
```
Comment (francislavoie) [inline cmd/commandfuncs.go:633]: What happens otherwise?

File: modules/caddyhttp/push/handler.go [hunk 3]
```diff
@@ -122,11 +122,8 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddyhtt
 	}
 
 	// serve only after pushing!
-	if err := next.ServeHTTP(lp, r); err != nil {
-		return err
-	}
-
-	return nil
+	err := next.ServeHTTP(lp, r)
+	return err
```
Comment (francislavoie) [inline modules/caddyhttp/push/handler.go:126]: I guess this can be simplified even further to `return next.ServeHTTP(lp, r)`

File: modules/caddyhttp/reverseproxy/selectionpolicies.go [hunk 4]
```diff
@@ -770,7 +770,7 @@ func leastRequests(upstreams []*Upstream) *Upstream {
 		return nil
 	}
 	var best []*Upstream
-	var bestReqs int = -1
+	bestReqs := -1
```
Comment (francislavoie) [inline modules/caddyhttp/reverseproxy/selectionpolicies.go:773]: Nitpick but this is actually less symmetrical so it looks less nice. But whatever, it's fine.

File: modules/caddyhttp/fileserver/matcher.go [hunk 5]
```diff
@@ -181,15 +181,15 @@ func (MatchFile) CELLibrary(ctx caddy.Context) (cel.Library, error) {
 			root = values["root"][0]
 		}
 
-		var try_policy string
+		var tryPolicy string
 		if len(values["try_policy"]) > 0 {
 			root = values["try_policy"][0]
```
Comment (francislavoie) [inline modules/caddyhttp/fileserver/matcher.go:186]: Oh I think there's actually a bug here, this should be `tryPolicy = values["try_policy"][0]`

File: modules/logging/nopencoder.go [hunk 6]
```diff
@@ -26,87 +26,87 @@ type nopEncoder struct{}
 
 // AddArray is part of the zapcore.ObjectEncoder interface.
 // Array elements do not get filtered.
-func (nopEncoder) AddArray(key string, marshaler zapcore.ArrayMarshaler) error { return nil }
+func (nopEncoder) AddArray(_ string, _ zapcore.ArrayMarshaler) error { return nil }
```
Comment (francislavoie) [inline modules/logging/nopencoder.go:29]: I'm not sure I like this. It makes it harder to read the intent of the params, even if they aren't used _for now_.

File: modules/caddyhttp/ip_range.go (hunk 1 above)
Comment (faddat) [reply modules/caddyhttp/ip_range.go]: f is for caddyfile, and it's used elsewhere on this receiver.  If you turn the revive linter on in the present state, it will flag this.

File: cmd/commandfuncs.go (hunk 2 above)
Comment (faddat) [reply cmd/commandfuncs.go:633]: the revive linter prompts us to remove the empty code block that is currently just comments.

File: modules/logging/nopencoder.go (hunk 6 above)
Comment (faddat) [reply modules/logging/nopencoder.go:29]: in fact, I also am not sure I like it.  Last time I did a large lint of a repo, I chose not to use this linter.  Would you like me to disable it?

Comment (mholt) [reply modules/logging/nopencoder.go:29]: Disagreements with linters is why we've not used a lot of them yet, I think :sweat_smile:

If we have ways of disabling certain checks, maybe that's manageable.

**Comment by faddat** [comment]
It's kinda my specialty and I am very happy to customize the configuration
to suit Caddy's unique needs.

File: modules/logging/nopencoder.go (hunk 6 above)
Comment (faddat) [reply modules/logging/nopencoder.go:29]: Configuring linters kinda one of my specialties and I am very happy to customize the configuration to suit Caddy's unique needs.

In this case I put those unused parameter lint fixes last so they're easy to revert because I linted github.com/cosmos/ibc-go recently and came to the same conclusion that you did :D

File: modules/caddyhttp/fileserver/matcher.go (hunk 5 above)
Comment (faddat) [reply modules/caddyhttp/fileserver/matcher.go]: does this commit address the issue?

https://github.com/caddyserver/caddy/pull/5712/commits/6f1286a630f226e5f679db3f6cd82b55cdc9e31b

**Comment by francislavoie** [comment]
I'll review again after the parent PRs are merged, hard to follow the diffs until then.

**Comment by faddat** [comment]
Sorry, this one should be on draft for a minute.  I may have revive configured a little too spicy.

**Comment by faddat** [comment]
I think this PR is no longer needed.  

Prefer #5709
